ios/Classes/SwiftFlutterVlcPlayerPlugin.swift

```diff
@@ -0,0 +1,284 @@
+                    result(nil)
+                    return
+                    
+                case .setTime:
+                    
+                    let time = VLCTime(number: arguments["time"] as? NSNumber)
+                    self.player.time = time
+                    result(nil)
+                    return
+                    
+                default:
+                    result(FlutterMethodNotImplemented)
+                    return
+                }
+            }
+            
+        })
+        
+        eventChannel.setStreamHandler(eventChannelHandler)
+        return hostedView
+        
+    }
+    
+    
+    
+}
+
+class VLCPlayerEventStreamHandler:NSObject, FlutterStreamHandler, VLCMediaPlayerDelegate {
+    
+    private var eventSink: FlutterEventSink?
+    
+    
+    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
+        self.eventSink = events
+        return nil
+    }
+    
+    func onCancel(withArguments arguments: Any?) -> FlutterError? {
+        eventSink = nil
+        return nil
+    }
+    
+    
+    func mediaPlayerStateChanged(_ aNotification: Notification?) {
+        
+        let player = aNotification?.object as? VLCMediaPlayer
+        let media = player?.media
+        let tracks: [Any] = media?.tracksInformation ?? [""]  //[Any]
+        var track:NSDictionary
+        
+        var ratio = Float(0.0)
+        var height = 0
+        var width =  0
+        
+        if player?.currentVideoTrackIndex != -1 {
+            if (player?.currentVideoTrackIndex) != nil {
+                track =  tracks[0] as! NSDictionary
+                height = track["height"] as! Int
+                width = track["width"] as! Int
+                
+                if height != 0 && width != 0  {
+                    ratio = Float(width / height)
+                }
+                
+            }
+            
+        }
+        
+        switch player?.state {
+        case .esAdded, .buffering, .opening:
+            return
+        case .playing:
+            eventSink?([
+                "name": "buffering",
+                "value": NSNumber(value: false)
+            ])
+            if let value = media?.length.value {
+                eventSink?([
+                    "name": "playing",
+                    "value": NSNumber(value: true),
+                    "ratio": NSNumber(value: ratio),
+                    "height": height,
+                    "width": width,
+                    "length": value
+                ])
+            }
+            return
+        case .ended:
+            eventSink?([
+                "name": "ended"
+            ])
+            eventSink?([
+                "name": "playing",
+                "value": NSNumber(value: false),
+                "reason": "EndReached"
+            ])
+            return
+        case .error:
+            print("(flutter_vlc_plugin) A VLC error occurred.")
```

Comment:
What do you think of returning an error to dart? 

```
case .error:
eventSink(FlutterError(code: "500",
                             message: "Player state got an error",
                             details: nil))
```

Reply:
Yea I wanted to do this, just didnt know exactly how. Do you have a more concrete example? Love the idea

Reply:
I think we can look at some plugins developed by google. 

I guess this [question]( https://stackoverflow.com/questions/59693268/how-to-stream-data-from-swift-to-flutter-using-event-channel/59759074#59759074) might help, I tried to found an example by google packages, however almost all of them are written in Obj-C. I hope it helps.

@mitchross